Add task/waiting/time metric (#18735)
druid-io/druid: docs/operations/metrics.md

|Metric|Description|Dimensions|Normal value|
|------|-----------|----------|------------|
|`task/run/time`|Milliseconds taken to run a task.| `dataSource`, `taskId`, `taskType`, `groupId`, `taskStatus`, `description`, `tags`|Varies|

> |`task/pending/time`|Milliseconds taken for a task to be scheduled and start running.| `dataSource`, `taskId`, `taskType`, `groupId`, `tags`|Varies|

Reviewer: ```suggestion
|`task/pending/time`|Milliseconds taken for a task to start running after being scheduled by the Overlord.| `dataSource`, `taskId`, `taskType`, `groupId`, `tags`|Varies|
```
